src/main/java/org/wso2/carbon/esb/connector/MQConnectionBuilder.java

```diff
@@ -0,0 +1,218 @@
+        }
+        MQEnvironment.properties.put(CMQC.TRANSPORT_PROPERTY, CMQC.TRANSPORT_MQSERIES_CLIENT);
+        MQEnvironment.properties.put(CMQC.USER_ID_PROPERTY, config.getUserName());
+        MQEnvironment.properties.put(CMQC.PASSWORD_PROPERTY, config.getPassword());
+        Collection headerComp = new Vector();
+        headerComp.add(new Integer(CMQXC.MQCOMPRESS_SYSTEM));
+        MQEnvironment.hdrCompList = headerComp;
+        MQEnvironment.setDefaultConnectionManager(customizedPool(config.getTimeout(), config.getmaxConnections(), config.getmaxnusedConnections()));
+
+        List<String> hostandportList;
+        List<String> channelList;
+
+        if (config.getReconnectList() != null) {
+            hostandportList = config.getReconnectList();
+        } else {
+            hostandportList = new ArrayList<>();
+        }
+
+        if (config.getChannelList() != null) {
+            channelList = config.getChannelList();
+        } else {
+            channelList = new ArrayList<>();
+        }
+
+        hostandportList.add(0, config.getHost() + "/" + config.getPort());
+        channelList.add(0, config.getChannel());
+
+        if (queueManager == null || !queueManager.isConnected()) {
+            queueManager = getQueueManager(config.getHost(), config.getChannel(), config.getPort());
+            if (queueManager == null) {
+
+                long start = System.currentTimeMillis();
+                long end = start + config.getReconnectTimeout() * 1000;
+                A:
+                while (System.currentTimeMillis() < end) {
+                    for (String hostandport : hostandportList) {
+                        String[] hostandportArray = hostandport.split("/");
+                        for (String channel : channelList) {
+                            queueManager = getQueueManager(hostandportArray[0], channel, Integer.valueOf(hostandportArray[1]));
+                            if (queueManager != null) {
+                                break A;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Initialize queue manager
+     */
+    public MQQueueManager getQueueManager(String host, String channel, int port) {
+
+        MQEnvironment.hostname = host;
+        MQEnvironment.channel = channel;
+        MQEnvironment.port = port;
+
+        Future<String> initializeQmanager
+                = Executors.newSingleThreadExecutor().submit(() -> {
+            try {
+                queueManager = new MQQueueManager(config.getqManger());
+                return "Initialized";
+            } catch (MQException e) {
+            }
+            return null;
+        });
+        try {
+            initializeQmanager.get(2, TimeUnit.SECONDS);
+            logger.info("Queue manager connection established for " + host + " " + port + " " + channel);
+        } catch (InterruptedException e) {
+            return null;
```

Comment:
Please use proper logging for these cases, if logging is not needed you may add comment saying this is expected scenario and we will return 'null', 'null' will be properly handled at higher layer.